File: AccelerometerSensor/app/src/main/java/com/darshandangar/accelerometersensor/MainActivity.java
package com.darshandangar.accelerometersensor;

import androidx.appcompat.app.AppCompatActivity;

import android.hardware.Sensor;
import android.hardware.SensorEvent;
import android.hardware.SensorEventListener;
import android.hardware.SensorManager;
import android.os.Bundle;
import android.widget.TextView;
import android.widget.Toast;

import javax.sql.StatementEventListener;

public class MainActivity extends AppCompatActivity implements SensorEventListener {

    TextView taxvalue;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        taxvalue = findViewById(R.id.taxValue);

        SensorManager sensorManager = (SensorManager) getSystemService(SENSOR_SERVICE);

        if (sensorManager!=null) {
            Sensor accelerosensor = sensorManager.getDefaultSensor(Sensor.TYPE_ACCELEROMETER);
            if (accelerosensor!=null) {
                sensorManager.registerListener(this,accelerosensor,SensorManager.SENSOR_DELAY_NORMAL);
            }

        } else{
            Toast.makeText(this, "Sensor service not detected", Toast.LENGTH_SHORT).show();
        }


    }

    @Override
    public void onSensorChanged(SensorEvent sensorEvent) {
        if (sensorEvent.sensor.getType()==Sensor.TYPE_ACCELEROMETER){
            taxvalue.setText("X:" + sensorEvent.values[0] + "Y:" + sensorEvent.values[1] +"Z:" + sensorEvent.values[2]);
        }

    }

    @Override
    public void onAccuracyChanged(Sensor sensor, int i) {

    }
}

File: MemesApp/app/src/main/java/com/darshandangar/memesapp/MainActivity.kt
package com.darshandangar.memesapp

import android.content.Intent
import android.graphics.drawable.Drawable
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
//import kotlinx.android.synthetic.main.activity_main.*
import android.view.View
import android.widget.ImageView
import android.widget.ProgressBar
import android.widget.TextView
import android.widget.Toast
import com.android.volley.Request
import com.android.volley.RequestQueue
import com.android.volley.Response
import com.android.volley.toolbox.*
import com.bumptech.glide.Glide
import com.bumptech.glide.load.DataSource
import com.bumptech.glide.load.engine.GlideException
import com.bumptech.glide.request.RequestListener
import com.bumptech.glide.request.target.Target


class MainActivity : AppCompatActivity() {

    lateinit var imageView: ImageView
    lateinit var progressBar: ProgressBar

    //var currentImgUrl: String? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        imageView = findViewById(R.id.imageView)
        progressBar = findViewById(R.id.progressbar)

        loadmeme()
    }

    fun loadmeme(){
        progressBar.visibility = View.VISIBLE


        // Instantiate the cache
        val cache = DiskBasedCache(cacheDir, 1024 * 1024) // 1MB cap

// Set up the network to use HttpURLConnection as the HTTP client.
        val network = BasicNetwork(HurlStack())

// Instantiate the RequestQueue with the cache and network. Start the queue.
        val requestQueue = RequestQueue(cache, network).apply {
            start()
        }

        val url = "https://meme-api.com/gimme"

// Formulate the request and handle the response.
        val jsonObjectRequest = JsonObjectRequest(
            Request.Method.GET, url, null,
            { response ->
                // Do something with the response
                val url = response.getString("url")
                Glide.with(this)
                    .load(url)
                    .listener(object: RequestListener<Drawable> {
                        override fun onLoadFailed(
                            e: GlideException?,
                            model: Any?,
                            target: Target<Drawable>?,
                            isFirstResource: Boolean
                        ): Boolean {
                            //TODO("Not yet implemented")
                            progressBar.visibility = View.GONE
                            return false

                        }

                        override fun onResourceReady(
                            resource: Drawable?,
                            model: Any?,
                            target: Target<Drawable>?,
                            dataSource: DataSource?,
                            isFirstResource: Boolean
                        ): Boolean {
                           //TODO("Not yet implemented")
                            progressBar.visibility = View.GONE
                            return false

                        }
                    })
                    .into(imageView)


            },
            { error ->
                // Handle error
                Toast.makeText(this,"Something went wrong",Toast.LENGTH_LONG).show()
            })

// Add the request to the RequestQueue.
        requestQueue.add(jsonObjectRequest)
    }
    fun shareMeme(view: View) {
        val intent = Intent(Intent.ACTION_SEND)
        intent.putExtra(Intent.EXTRA_TEXT,"Hey,checkout this cool meme I got from Reddit ")
        val chooser = Intent.createChooser(intent,"Share this Meme using...")
        startActivity(chooser)
    }

    fun nextMeme(view: View) {
        loadmeme()
    }
}


